### PR TITLE
search counter issue solved

### LINKED ISSUE #1736: Search number is not changing
**Actual Behaviour**

Search number is not changing

**Expected Behaviour**

Search number should change correctly

**Screenshots of the issue**

search

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by prateek76**
Fixes #1736  @uday96 @akshatnitd @anshumanv @DNS-404  please review

Changes:
fixed not working counter in search counter

Demo Link: https://pr-1815-fossasia-susi-web-chat.surge.sh/
gif-
peekai

### CODE REVIEW

**Comment by prateek76** [comment]
@rajvaibhavdubey I have edited the description please review

**Comment by prateek76** [comment]
@rajvaibhavdubey it's not showing merge conflicts to me, how should i remove it?

**Comment by prateek76** [comment]
@anshumanv @isuruAb please review

**Comment by AakashMallik** [comment]
@prateek76 Update your branch

**Comment by prateek76** [comment]
@AakashMallik updated

File: src/components/ChatApp/SearchField.react.js [hunk 1]
```diff
@@ -138,7 +149,7 @@ class ExpandingSearchField extends Component {
             autoFocus={true}
           />
           <span className="counter">
-            {this.props.searchIndex}/{this.props.searchCount}
+            {this.state.indexCnt}/{this.props.searchCount}
           </span>
```
Comment (AakashMallik) [inline src/components/ChatApp/SearchField.react.js]: Use destructuring for props and states.

File: src/components/ChatApp/SearchField.react.js [hunk 2]
```diff
@@ -66,10 +69,18 @@ class ExpandingSearchField extends Component {
...
     this.props.scrollRecent();
+    this.setState({
+      indexCnt: (this.state.indexCnt % this.props.searchCount) + 1,
+    });
   };
 
   onClickPrev = () => {
     this.props.scrollPrev();
+    if (this.state.indexCnt > 1) {
+      this.setState({ indexCnt: this.state.indexCnt - 1 });
```
Comment (AakashMallik) [inline src/components/ChatApp/SearchField.react.js]: Destructure state.

File: src/components/ChatApp/SearchField.react.js [hunk 3]
```diff
@@ -66,10 +69,18 @@ class ExpandingSearchField extends Component {
...
   onClickRecent = () => {
     this.props.scrollRecent();
+    this.setState({
+      indexCnt: (this.state.indexCnt % this.props.searchCount) + 1,
+    });
   };
 
   onClickPrev = () => {
     this.props.scrollPrev();
+    if (this.state.indexCnt > 1) {
```
Comment (AakashMallik) [inline src/components/ChatApp/SearchField.react.js]: Destructure

File: src/components/ChatApp/SearchField.react.js [hunk 4]
```diff
@@ -66,10 +69,18 @@ class ExpandingSearchField extends Component {
 
   onClickRecent = () => {
     this.props.scrollRecent();
+    this.setState({
+      indexCnt: (this.state.indexCnt % this.props.searchCount) + 1,
```
Comment (AakashMallik) [inline src/components/ChatApp/SearchField.react.js]: Destructure

**Comment by AakashMallik** [comment]
@prateek76 The down arrow cycles the search propperly but the up arrow doesn' t do so. Fix it please. Also squash your commits when you finalize your PR.

**Comment by prateek76** [comment]
@AakashMallik what is the problem with using this.state it is used in many places in the project

**Comment by AakashMallik** [comment]
> @AakashMallik what is the problem with using this.state it is used in many places in the project

@prateek76 One good reason for not using **this.state** would be coz it is bad practice. And we have used destrucrting in every component that we have refactored. So I think to make your work more consistent with the rest of the code base, please destructure the state variables before using it.

**Comment by prateek76** [comment]
@AakashMallik up arrow search cycle is not searching in up direction so counter is not updating, issue was to make counter functional i did that

**Comment by prateek76** [comment]
> > @AakashMallik what is the problem with using this.state it is used in many places in the project
> 
> @prateek76 One good reason for not using **this.state** would be coz it is bad practice. And we have used destrucrting in every component that we have refactored. So I think to make your work more consistent with the rest of the code base, please destructure the state variables before using it.

ok, thanks i will do it

**Comment by prateek76** [comment]
@AakashMallik  please review, i have destructured the state variables, but i am not able to fix uparrow counter cycle and it was also not part of this issue

File: src/components/ChatApp/SearchField.react.js [hunk 5]
```diff
@@ -64,12 +67,26 @@ class ExpandingSearchField extends Component {
...
+    });
   };
 
-  onClickPrev = () => {
+  onClickPrev = (state, props) => {
+    const { indexCnt } = this.state;
+
     this.props.scrollPrev();
+    if (indexCnt > 1) {
+      this.setState({ indexCnt: indexCnt - 1 });
```
Comment (AakashMallik) [inline src/components/ChatApp/SearchField.react.js]: To fix the up arrow issue make changes here, 
Hint: If indexCnt goes below a certain value, change it back to searchCount. Some modification might be need but it should be pretty straight forward.

**Comment by AakashMallik** [comment]
@prateek76 I have given you a hint to fix the down arrow issue. Fix it and we should be good to go.

File: src/components/ChatApp/SearchField.react.js (hunk 5 above)
Comment (prateek76) [reply src/components/ChatApp/SearchField.react.js]: i got what you want but the problem is the counter value is changing but its not highlighting the words when i press the uparrow, it has some other issue

Comment (prateek76) [reply src/components/ChatApp/SearchField.react.js]: i got a fix

**Comment by prateek76** [comment]
@AakashMallik  done the changes please review

**Comment by AakashMallik** [comment]
screenshot-pr-1815-fossasia-susi-web-chat surge sh-2018 12 11-20-36-41

Don't show the numbers if the text field is empty.

**Comment by prateek76** [comment]
@AakashMallik done please review, ThankYou

**Comment by AakashMallik** [comment]
Just update the branch and rest look good to me. Nice work buddy.

**Comment by prateek76** [comment]
@AakashMallik please merge the pr

**Comment by prateek76** [comment]
@AakashMallik  @akshatnitd  please merge the pr

**Comment by prateek76** [comment]
@AakashMallik does not passing a check makes any problem?

**Comment by AakashMallik** [comment]
> @AakashMallik does not passing a check makes any problem?

Nah, it is okay. It must pass CI, that is a must. Codacy is usually igonred if you have minor warning. @prateek76

**Comment by prateek76** [comment]
> > @AakashMallik does not passing a check makes any problem?
> 
> Nah, it is okay. It must pass CI, that is a must. Codacy is usually igonred if you have minor warning. @prateek76

thanks, does it take much time to merge the pr

**Comment by prateek76** [comment]
@anshumanv fixed it please review

**Comment by anshumanv** [comment]
image

There are still cases when 0 shows up, handle those, press enter once you type something in the search field.

**Comment by prateek76** [comment]
@anshumanv i made a mess while squashing the commits should i make a new pr please help

**Comment by prateek76** [comment]
@akshatnitd please review

**Comment by prateek76** [comment]
@samagragupta can you merge the pr

**Comment by prateek76** [comment]
@rajvaibhavdubey what are talking about? it's not showing any conflicts

**Pull request merged**